### DISCUSSION TITLE
`winget source update` should not show "Cancelled"

### DISCUSSION THREAD

**Discussion by nopeless** [microsoft/winget-cli#4183] [Ideas] [answered]
"Cancelled" gives no indication of what went wrong

**Comment by davidldennison** [comment]
Been running into this as well

**Comment by denelon** [comment]
Could you share which version of WinGet you're seeing this on as well as verbose logs?

**Reply by nopeless** [reply]
I hate to say this but I have already moved on from this issue and have a fresh install. Additionally, I find it hard to believe that complete logs are require to give more information than just "cancelled" from the winget update command. For example, failed appx actions logs appropriate commands and links for the developer/admin to further troubleshoot the issue.

In my case, the cause of the issue was an unrepairable corrupted appx state repository. I believe this is a rather rare case of it happening because almost nobody had this issue and realized I would not get any support for it. If you want the minified disk image copy (I haven't fully migrated my system) for testing those winget commands in appx and actually forwarding the issue to whatevers department develops appx interface let me know.

TLDR: A. I unfortunately couldn't care more about this issue any more B. This issue doesn't necessarily apply to my issue, but any sort of failures that might be encountered. A helpful guide such as "look into eventvwr with someField X" would be more helpful than "cancelled". C. if you care about fixing the root cause of my issue, which is an Appx one, not winget, you need a copy of my OS. let me know if you want this

**Reply by denelon** [reply, marked as answer]
OK, the most important thing is that you're not blocked. If someone else encounters the same issue, I'll be able to troubleshoot. One of the most recent changes we made in the client was to address database corruption.

**Reply by nopeless** [reply]
> OK, the most important thing is that you're not blocked. If someone else encounters the same issue, I'll be able to troubleshoot. One of the most recent changes we made in the client was to address database corruption.

Additionally, I would like to see some documentation about how appx works internally. I tried to look into it and there were just so many tables and entries that I found it impossible to fix it on my own. Nevertheless, thanks for continuing to improve winget. I have high hopes for it

**Reply by denelon** [reply]
I'm assuming you've already looked at https://learn.microsoft.com/windows/msix/overview where everything "should" be documented.

**Reply by denelon** [reply]
This covers the "lifecycle" https://learn.microsoft.com/windows/msix/desktop/deploy-preinstalled-apps

**Reply by nopeless** [reply]
It was an Appx issue that spanned across multiple packages. The lifecycle would be irrelevant. https://github.com/microsoft/winget-cli/issues/4190

I have tried a lot of ways and all of them seem to not get past some kind of check (they are logged in event viewer). I am not sure what the check does exactly but I am sure that it applies to duplicate packages in the Appx state repository. This is done at the dll level and starting from here it was getting exponentially harder to troubleshoot.

When I was asking for documentation, I was specifically looking for documentations about the appx deployment dll and `C:\ProgramData\Microsoft\Windows\AppRepository\StateRepository-*.srd`. I have reached out to one windows sysadmin I know and his best guess was that this is all private source.

I hope this clarifies what kind of efforts I went through and what my exact interests are.
